fix kind job with network policy failures

aojea opened the pull request:
The github runners do not have enough resources to run multiple network policy tests in parallel and this cause the test to freeze and create flakiness, remove the legacy tests are deprecated and removed in kubernetes 1.28 and only run 5 tests in parallel to avoid problems caused by resource starvation on CI.


Example of resource usage with this PR 

https://github.com/cilium/cilium/actions/runs/5518744069?pr=26639

Example of resource usage without it

https://github.com/cilium/cilium/actions/runs/5513171662



xref: https://github.com/kubernetes/kubernetes/pull/118915

```release-note
```

Fixes: #26439,#26492

aanm reviewed (commented):
Should we remove them already given that we are still in k8s 1.27?

aojea committed:
disable legacy network policy tests (deprecated)

Signed-off-by: Antonio Ojea <aojea@google.com>

aojea committed:
reduce network policy parallelism to avoid starving the runner

Github runners does not have enough resources to deal with the
network policy test that runs multiple pods in parallal and
it may happen that the tests get stuck and start to timeout
causing flakiness on the CI.

Signed-off-by: Antonio Ojea <aojea@google.com>
